### PR TITLE
don't emit #[repr(align(0))], and reduced test case

### LINKED ISSUE #1565: repr(align(0)) for std::basic_string causing build error
### Input C/C++ Header

```C++
#include <string>
using namespace std;
basic_string<char> simple(basic_string<char> str);
```

### Bindgen Invocation

```Rust
bindgen::Builder::default()
    .header("simple.hpp")
    .clang_arg("-x")
    .clang_arg("c++")
    .opaque_type("std::.*")
    .whitelist_function("simple")
    .generate()
    .unwrap()
```

### Actual Results

Generating the binding.rs works fine, but when compiling....
```
cargo build
  Compiling simple-bindgen v0.1.0 (simple-bindgen)
error[E0589]: invalid `repr(align)` attribute: not a power of two
 --> simple-bindgen/target/debug/build/simple-bindgen-0af4d98c1d10abb2/out/bindings.rs:73:8
  |
73 | #[repr(align(0))]
  |        ^^^^^^^^

error[E0589]: invalid `repr(align)` attribute: not a power of two
 --> simple-bindgen/target/debug/build/simple-bindgen-0af4d98c1d10abb2/out/bindings.rs:90:8
  |
90 | #[repr(align(0))]
  |        ^^^^^^^^
```
The offending lines in the binding.rs look like:

```rust
#[repr(C)]
#[repr(align(0))]
pub union std_basic_string___ulx {
    pub _bindgen_opaque_blob: [u8; 0usize],
}
```

and

```rust
#[repr(C)]
#[repr(align(0))]
pub union std_basic_string___rep__bindgen_ty_1 {
    pub _bindgen_opaque_blob: [u8; 0usize],
}
```

### Expected Results

While this works just fine on our linux machines, on our Macs, we get the error provided. Here's a sample `uname -a` from one of our Macs:

```
Darwin computer.local 18.5.0 Darwin Kernel Version 18.5.0: Mon Mar 11 20:40:32 PDT 2019; root:xnu-4903.251.3~3/RELEASE_X86_64 x86_64
```

### PR DESCRIPTION
**Pull request by pmarks**
fixes #1565 
not sure if this is the right fix, but the test case should be useful.

### CODE REVIEW

File: tests/headers/repr-align.hpp [hunk 1]
```diff
@@ -9,3 +9,7 @@ struct alignas(double) b {
   int b;
   int c;
 };
+
+namespace std {
+union a;
```
Comment (emilio) [inline tests/headers/repr-align.hpp]: Oh, so it's a forward declaration, fun... Let me think a bit whether it's the right fix. Thank you so much for the test-case!

**Comment by pmarks** [comment]
@emilio thanks for your help! The PR now fixes the original problem, but introduces the following test failure, which unfortunately is way over my head. LMK if you have ideas on how to proceed.

```
---- header_objc_template_h stdout ----



diff expected generated
--- expected: "/Users/patrick/code/rust-bindgen/tests/expectations/tests/libclang-5/objc_template.rs"
+++ generated from: "/Users/patrick/code/rust-bindgen/tests/headers/objc_template.h"
 /* automatically generated by rust-bindgen */

 #![allow(
     dead_code,
     non_snake_case,
     non_camel_case_types,
     non_upper_case_globals
 )]
 #![cfg(target_os = "macos")]

 #[macro_use]
 extern crate objc;
 #[allow(non_camel_case_types)]
 pub type id = *mut objc::runtime::Object;
 pub trait Foo {
-    unsafe fn get(self) -> *mut ObjectType;
+    unsafe fn get(self) -> u64;
 }
 impl Foo for id {
-    unsafe fn get(self) -> *mut ObjectType {
+    unsafe fn get(self) -> u64 {
         msg_send!(self, get)
     }
 }
```

File: tests/headers/repr-align.hpp [hunk 2]
```diff
@@ -9,3 +9,4 @@ struct alignas(double) b {
   int b;
   int c;
 };
+
```
Comment (emilio) [inline tests/headers/repr-align.hpp]: nit: remove this addition too?

**Comment by emilio** [comment]
And to be clear, that failure should happen with on without your fix, I'm pretty sure.

**Comment by pmarks** [comment]
@emilio apologies: I didn't recheck the original test case.  It's still broken.  I'm re-reducing w/ the first fix in place.

**Comment by pmarks** [comment]
Here's the new test case:

**Comment by emilio** [comment]
Ok, so this looks good to merge as-is. I'll suggest a fix for the second test-case as well. But TL;DR: it should be in the same function. Empty unions don't have zero alignment, but we're coming up with that in that test-case.

**Pull request merged**